### ISSUE TITLE
Doesn't find the files with eslintignore is not in the cwd and cwd path has special character

### ISSUE THREAD

**Issue by Teamop** [eslint/eslint#11087]
**Tell us about your environment**

* **ESLint Version:** master branch
* **Node Version:** 8.11.4
* **npm Version:** 6.4.1
* **OS:** win 10 

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**
Just use the master branch

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I found this issue when I tried to run the unit tests on the master branch of the eslint repo itself. There are two cases cannot pass. But I checked the status in the repo seems all the test cases can pass without any error.
Error test cases are all under `executeOnFiles()` of `cli-engine.js`:
1. `should throw an error when all given files are ignored`
1. `should throw an error when all given files are ignored even with a './' prefix`
![image](https://user-images.githubusercontent.com/8519685/48537242-f0389a00-e8eb-11e8-82b1-9eeeb76122f1.png)

Then I did some research. I found the issue is because:
1. the current `.eslintignore` file has the rule `*.js`
2. the current cwd path has the special character `.js`

You can see from the above image, the cwd path is: `C:\Work\Node.js\workspace\OSS\eslint`, so here the path contains `Node.js` which matches the rule in the ignore file. When I change the cwd path to `C:\Work\Node\workspace\OSS\eslint`, then all test cases can pass.

Also from the code perspective, 
```
// code from the ignored-paths.js source code
return function(absolutePath) {
            const relative = pathUtils.getRelativePath(absolutePath, baseDir);

            if (!relative) {
                return false;
            }

            return !filter(relative);
};

```
Here the `absolutePath` is  `C:/Work/Node.js/workspace/OSS/eslint/tests/fixtures`, the `baseDir` is `C:\` because of the ignore path is OS tmp path. So for the filter, the `relative` is `Work/Node.js/workspace/OSS/eslint/tests/fixtures`, so for the ignore rule `*.js`, it matches. Then these files are ignored, so get `NoFilesFoundError` instead of `AllFilesIgnoredError`

```js
// so here the easier way to reproduce the error with the special cwd
node bin\eslint --ignore-path %USERPROFILE%\AppData\Local\Temp\eslint\fixtures\.eslintignore tools\*.js
```

**What did you expect to happen?**
The test cases should pass

**What actually happened? Please include the actual, raw output from ESLint.**
See the above

**Are you willing to submit a pull request to fix this bug?**

Yes, but I haven't found any solutions yet.

**Comment by kaicataldo** [comment]
I think this is a duplicate of https://github.com/eslint/eslint/issues/10821.

**Comment by Teamop** [comment]
@kaicataldo No, I don't think it's a duplicate of #10821. You can see from the detail, here the special character is the one that will match the rule from `.eslintignore`, while the one in #10821 is from other language. Also, even that one also mention a [base directory issue](https://github.com/eslint/eslint/issues/10821#issuecomment-438375539), but that's when updating `ignore` to the v5 version.
